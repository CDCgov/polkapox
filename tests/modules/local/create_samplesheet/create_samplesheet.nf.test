nextflow_process {

    name "Test Process CREATE_SAMPLESHEET"
    script "modules/local/create_samplesheet/create_samplesheet.nf"
    process "CREATE_SAMPLESHEET"

    test("Test | paired-end reads") {

        when {
            params {
                project_name = "Test"
                paired = true
                outdir = ${outputDir}
            }
            process {
                """
                // input[0] = file("path/to/input/dir")
                input[0] = file("/scicomp/groups-pure/Projects/scbs_mpob/JRowell_ick4/mpox_testing/paired/")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Test | single-end reads") {

        when {
            params {
                project_name = "Test"
                paired = false
                outdir = ${outputDir}
            }
            process {
                """
                // input[0] = file("path/to/input/dir")
                input[0] = file("/scicomp/groups-pure/Projects/scbs_mpob/JRowell_ick4/mpox_testing/single/")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }

}
