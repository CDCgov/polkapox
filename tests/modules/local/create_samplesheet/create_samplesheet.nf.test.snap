{
    "Test | single-end reads": {
        "content": [
            {
                "0": [
                    "Test_samplesheet.csv:md5,6d78c92cb2ea8938e0541a32c62a7d1e"
                ],
                "1": [
                    "versions.yml:md5,29c636ba5752db4630100b2d9c5e8b95"
                ],
                "samplesheet": [
                    "Test_samplesheet.csv:md5,6d78c92cb2ea8938e0541a32c62a7d1e"
                ],
                "versions": [
                    "versions.yml:md5,29c636ba5752db4630100b2d9c5e8b95"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-06-24T12:23:15.588289245"
    },
    "Test | paired-end reads": {
        "content": [
            {
                "0": [
                    "Test_samplesheet.csv:md5,462af9491a3b144c812aac6f24510382"
                ],
                "1": [
                    "versions.yml:md5,29c636ba5752db4630100b2d9c5e8b95"
                ],
                "samplesheet": [
                    "Test_samplesheet.csv:md5,462af9491a3b144c812aac6f24510382"
                ],
                "versions": [
                    "versions.yml:md5,29c636ba5752db4630100b2d9c5e8b95"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-06-24T12:23:09.869779873"
    }
}